### PR TITLE
Feature/11565 create fss mock api endpoint for create batch upload and write blocks to upload zip with commit batch

### PR DESCRIPTION
**Pull request by ganpat-tech**
PR contains created FSS mock API endpoints for Create Batch, Upload and Write Blocks ,Commit Batch to upload zip.

### CODE REVIEW

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Filters/BinaryRequestBodyFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        public BinaryRequestBodyFormatter()
+        {
+            SupportedMediaTypes.Add(new MediaTypeHeaderValue("application/octet-stream"));
+        }
+
+        public override bool CanRead(InputFormatterContext context)
+        {
+            if (context == null) throw new ArgumentNullException(nameof(context));
+
+            var contentType = context.HttpContext.Request.ContentType;
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Filters/BinaryRequestBodyFormatter.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 2]
```diff
@@ -50,16 +72,75 @@ public IActionResult GetBatches([FromQuery] int? limit, [FromQuery] int start =
...
+        [Route("batch/{batchId}/files/{fileName}/{blockId}")]
+        [Produces("application/json")]
+        [Consumes("application/octet-stream")]
+        public IActionResult UploadBlockOfFile( [FromRoute, SwaggerSchema(Format = "GUID"), SwaggerParameter(Required = true)] string batchId,
+                                                           [FromRoute, SwaggerParameter(Required = true)] string fileName,
+                                                           [FromRoute, SwaggerParameter(Required = true)] string blockId,
+                                                           [FromHeader(Name = "Content-Length"), SwaggerSchema(Format = ""), SwaggerParameter(Required = true)] decimal? contentLength,
+                                                           [FromHeader(Name = "Content-MD5"), SwaggerSchema(Format = "byte"), SwaggerParameter(Required = true)] string contentMD5,
+                                                           [FromHeader(Name = "Content-Type"), SwaggerSchema(Format = "MIME"), SwaggerParameter(Required = true)] string contentType,
+                                                           [FromBody] object data )
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 3]
```diff
@@ -50,16 +72,75 @@ public IActionResult GetBatches([FromQuery] int? limit, [FromQuery] int start =
...
+            return BadRequest();
+        }
+
+        [HttpPut]
+        [Route("batch/{batchId}/files/{fileName}")]
+        [Produces("application/json")]
+        [Consumes("application/json")]
+        public IActionResult PutBlocksInFile([FromRoute, SwaggerSchema(Format = "GUID"), SwaggerParameter(Required = true)] string batchId,
+                                                               [FromRoute, SwaggerParameter(Required = true)] string fileName,
+                                                               [FromBody, SwaggerParameter(Required = true)] FileCommitPayload payload)
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 4]
```diff
@@ -12,24 +16,42 @@ public class FileShareServiceController : BaseController
...
+        public IActionResult CreateBatch([FromBody] BatchRequest batchRequest)
         {
-            if (!string.IsNullOrEmpty(correlationId))
+            if (batchRequest != null && !string.IsNullOrEmpty(batchRequest.BusinessUnit))
             {
-                return Ok(new CreateBatchResponse());
+                var response = fileShareService.CreateBatch(configuration["HOME"]);
+                if (response != null)
+                {
+                    return Created("", response);
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

**Review by barrie-cooper — APPROVED** [review]
LGTM

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**